File: 2038-remove-colored-pieces-if-both-neighbors-are-the-same-color/2038-remove-colored-pieces-if-both-neighbors-are-the-same-color.java
class Solution {
    public boolean winnerOfGame(String colors) {
        int alice_plays = 0, bob_plays = 0, count = 0;
        
        for (int i = 1; i < colors.length(); i++) {
            if (colors.charAt(i) == colors.charAt(i - 1)) {
                count++;
            } else {
                if (colors.charAt(i - 1) == 'A') {
                    alice_plays += Math.max(0, count - 1);
                } else {
                    bob_plays += Math.max(0, count - 1);
                }
                count = 0;
            }
        }

        if (colors.charAt(colors.length() - 1) == 'A') {
            alice_plays += Math.max(0, count - 1);
        } else {
            bob_plays += Math.max(0, count - 1);
        }
        
        return alice_plays > bob_plays;        
    }
}

File: 2038-remove-colored-pieces-if-both-neighbors-are-the-same-color/2038-remove-colored-pieces-if-both-neighbors-are-the-same-color.kt
class Solution {
    fun winnerOfGame(colors: String): Boolean {
        var alice_plays = 0
        var bob_plays = 0
        var count = 0
        for (i in 1 until colors.length) {
          if (colors[i] == colors[i - 1]) {
            count++
          } else {
            if (colors[i - 1] == 'A') {
              alice_plays += Math.max(0, count - 1)
            } else {
              bob_plays += Math.max(0, count - 1)
            }
            count = 0
          }
        }
        if (colors[colors.length - 1] == 'A') {
          alice_plays += Math.max(0, count - 1)
        } else {
          bob_plays += Math.max(0, count - 1)
        }
        return alice_plays > bob_plays
    }
}